### DISCUSSION TITLE
Keyword `as const` on locale const

### DISCUSSION THREAD

**Discussion by lorenzotomasi** [js-fns/date-fns#3712] [v3 API ideas]
Why in locales `_lib` files, only when an Array is present, its present the `as const` keyword? 

Also, why are these constants not exported?

An example from `src/locale/it/_lib/localize/index.ts` with the keyword
```ts
const eraValues = {
  narrow: ["aC", "dC"] as const,
  abbreviated: ["a.C.", "d.C."] as const,
  wide: ["avanti Cristo", "dopo Cristo"] as const,
};
```

An example from `src/locale/it/_lib/localize/index.ts` without the keyword
```ts 
const dayPeriodValues = {
  narrow: {
    am: "m.",
    pm: "p.",
    midnight: "mezzanotte",
    noon: "mezzogiorno",
    morning: "mattina",
    afternoon: "pomeriggio",
    evening: "sera",
    night: "notte",
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "mezzanotte",
    noon: "mezzogiorno",
    morning: "mattina",
    afternoon: "pomeriggio",
    evening: "sera",
    night: "notte",
  },
  wide: {
    am: "AM",
    pm: "PM",
    midnight: "mezzanotte",
    noon: "mezzogiorno",
    morning: "mattina",
    afternoon: "pomeriggio",
    evening: "sera",
    night: "notte",
  },
};
```

**Comment by joaopedrodcf** [comment]
You mean what the `as const` syntax does ?

**Reply by lorenzotomasi** [reply]
Generate a nested readonly typed object if I’m not remembering right.
Btw my question was relative to the fact that only the array use that keyword, and the fact that this constants are not exported.
I can immagine someone with the necessity to translate only some part of a date will benefit from this exported const
